hypothesis-python/src/hypothesis/_strategies.py

```diff
@@ -1292,12 +1292,18 @@ def from_type(thing):
             # At runtime, `typing.NewType` returns an identity function rather
             # than an actual type, but we can check that for a possible match
             # and then read the magic attribute to unwrap it.
-            if (
-                hasattr(thing, "__supertype__")
-                and hasattr(typing, "NewType")
-                and isfunction(thing)
-                and getattr(thing, "__module__", 0) == "typing"
-            ):
+            if types.is_a_new_type(thing):
+                # Check if we have an explicitly registered strategy for this thing,
+                # resolve it so, and otherwise resolve as for the base type.
+                if thing in types._global_type_lookup:
+                    strategy = types._global_type_lookup[thing]
+                    if not isinstance(strategy, SearchStrategy):
+                        strategy = strategy(thing)  # type: ignore
```

Comment:
This `# type: ignore` comment is [failing a build](https://travis-ci.org/HypothesisWorks/hypothesis/jobs/515071299#L822), because it (like the next two) is not needed.